Src/GoogleApis/Apis/[Media]/Download/MediaDownloader.cs

```diff
@@ -206,6 +208,16 @@ public IDownloadProgress Download(string url, Stream stream)
                     using (var response = await service.HttpClient.SendAsync(request, cancellationToken).
                         ConfigureAwait(false))
                     {
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            var error = await service.DeserializeError(response).ConfigureAwait(false);
+                            throw new GoogleApiException(service.Name, error.ToString())

```

Comment:
1. What made you adding this part of code?
2. I think that we shouldn't throw the exception. We should do something like:
https://github.com/google/google-api-dotnet-client/blob/master/Src/GoogleApis/Apis/%5BMedia%5D/Download/MediaDownloader.cs#L252

So you can create an exception, but actually you should update the progress and return it. That's the way that we did it before (I'm not sure that it's the best one, but that's what we picked back in the time...)